src/onnx/parse_matmul.cpp

```diff
@@ -89,55 +160,40 @@ struct parse_matmul : op_parser<parse_matmul>
+
+            if(not is_quant_dot and args.size() > 2)
             {
-                auto l0_it = s0_lens.begin() + s0_lens.size() - 2;
-                std::vector<std::size_t> l0_broadcasted_lens(s0_lens.begin(), l0_it);
-                auto l1_it = s1_lens.begin() + s1_lens.size() - 2;
-                std::vector<std::size_t> l1_broadcasted_lens(s1_lens.begin(), l1_it);
-                auto output_lens =
-                    compute_broadcasted_lens(l0_broadcasted_lens, l1_broadcasted_lens);
-                l0_broadcasted_lens = output_lens;
-                l0_broadcasted_lens.insert(l0_broadcasted_lens.end(), l0_it, s0_lens.end());
-                l1_broadcasted_lens = output_lens;
-                l1_broadcasted_lens.insert(l1_broadcasted_lens.end(), l1_it, s1_lens.end());
-                if(s0_lens != l0_broadcasted_lens)
-                {
-                    ba0 = info.add_instruction(
-                        make_op("multibroadcast", {{"out_lens", l0_broadcasted_lens}}), a0);
-                }
-                if(s1_lens != l1_broadcasted_lens)
-                {
-                    ba1 = info.add_instruction(
-                        make_op("multibroadcast", {{"out_lens", l1_broadcasted_lens}}), a1);
-                }
+                MIGRAPHX_THROW("PARSE_MATMUL: Bias Args not supported for MatMul");
             }
 
-            // parse a_zero_point and b_zero_point values
-            if(args.size() > 2)
+            if(is_quant_dot and
+               (not contains(supported_types, ba0_type) or not contains(supported_types, ba1_type)))
             {
-                ba0 = info.add_instruction(
-                    make_op("convert", {{"target_type", migraphx::shape::float_type}}), ba0);
-
-                ba0 = info.add_common_op("sub", ba0, args[2]);
-                if(args.size() > 3)
-                {
-                    ba1 = info.add_instruction(
-                        make_op("convert", {{"target_type", migraphx::shape::float_type}}), ba1);
-                    ba1 = info.add_common_op("sub", ba1, args[3]);
-                }
-                dot_res = info.add_instruction(make_op("dot"), ba0, ba1);
-                dot_res = info.add_instruction(
-                    make_op("convert", {{"target_type", migraphx::shape::int32_type}}), dot_res);
+                MIGRAPHX_THROW("PARSE_MATMULINTEGER: Unsupported type");
             }
-            else
+
+            auto is_same_type = (ba0_type == ba1_type);
+
+            if(is_quant_dot and not is_same_type and (ba0_type == migraphx::shape::uint8_type))
             {
-                dot_res = info.add_instruction(make_op(opd.op_name), ba0, ba1);
+                ba0 = add_int8_shift(info, ba0);
             }
+
+            if(is_quant_dot and not is_same_type and (ba1_type == migraphx::shape::uint8_type))
+            {
+                ba1 = add_int8_shift(info, ba1);
+            }
```

Comment:
it would be better if you can simplify it like 
```
if(is_quant_dot and not is_same_type) {
if(ba1_type == migraphx::shape::uint8_type) {}
if(ba2_type == migrpahx::shape::uint8_type)()
}
```

Reply:
I simplified it this way  because tidy was giving me errors on complexity but I see your point. Less checks better

Reply:
Done